updated and cleaned up some documentation (#295)
cucapra/dahlia: docs/cheat.md

## Views

Views are used to represent different memory access patterns.
Once created, they are accessed transparently as arrays.

### Shrink Views
A shrink view reduces the banking of a memory by a integer factor.

```dahlia
let A: float[16 bank 8];
view sh = shrink[by 4];
for (let i = 0..8) unroll 2 {
  sh[i]; // OK: sh has 2 banks (8 / 4 = 2)
}
```

### Suffix and Prefix Views
These views let you take a prefix or a suffix of a memory as long as the banking factor of a memory
is a factor of the prefix / suffix. To enforce this restriction, the syntax of suffix/prefix is
`suffix M[by k * e]` where `k` must be an integer.

```dahlia
let A: float[6 bank 4];
for (let i = 0..4) {
  view s = suffix A[by 2 * i];
  ---
  view s = prefix A[by 2 * i];
}
```

### Shift Views

> These are generalized suffix / prefix views that lift the restriction that the banking factor

Reviewer: ```suggestion
These are generalized suffix/prefix views that lift the restriction that the banking factor
```

is a factor of the shifting factor. The syntax for this view is `shfit M[by e]` where `e` can be
any expression.

```dahlia
let A: float[6 bank 4];
for (let i = 0..4) {
  view s = shift A[by i * i];
}
```

### Split Views

> Split views transform  single dimension of a memory into a multi-dimensional memory.

Reviewer: ```suggestion
Split views transform one dimension of a memory into multiple dimensions.
```

The split factor must divide the size of the dimension that is being split.

```dahlia
let A: float[8];

> view split_A = split A[by 2]; // split_A has type float[4 bank 2][2 bank 2];

Reviewer: ```suggestion
view split_A = split A[by 2]; // split_A has type float[4 bank 2][2 bank 2]
```
